runtime/jcl/common/jdk_jfr_internal_JVM_common.cpp

```diff
@@ -44,88 +44,81 @@ Java_jdk_jfr_internal_JVM_registerNatives(JNIEnv *env, jclass clazz)
 }
 
 void JNICALL
-Java_jdk_jfr_internal_JVM_markChunkFinal(JNIEnv *env, jclass clazz)
+Java_jdk_jfr_internal_JVM_beginRecording(JNIEnv *env, jobject obj)
 {
 	// TODO: implementation
 }
 
-void JNICALL
-Java_jdk_jfr_internal_JVM_beginRecording(JNIEnv *env, jclass clazz)
+jlong JNICALL
+Java_jdk_jfr_internal_JVM_counterTime(JNIEnv *env, jclass clazz)
 {
 	// TODO: implementation
+	return 0;
 }
 
 jboolean JNICALL
-Java_jdk_jfr_internal_JVM_isRecording(JNIEnv *env, jclass clazz)
+Java_jdk_jfr_internal_JVM_emitEvent(JNIEnv *env, jobject obj, jlong eventTypeId, jlong timestamp,
+#if JAVA_SPEC_VERSION <= 17
+		jlong when
+#else /* JAVA_SPEC_VERSION <= 17 */
+		jlong periodicType
+#endif /* JAVA_SPEC_VERSION <= 17 */
+)
 {
 	// TODO: implementation
 	return JNI_FALSE;
 }
 
 void JNICALL
-Java_jdk_jfr_internal_JVM_endRecording(JNIEnv *env, jclass clazz)
-{
-	// TODO: implementation
-}
-
-jlong JNICALL
-Java_jdk_jfr_internal_JVM_counterTime(JNIEnv *env, jclass clazz)
-{
-	// TODO: implementation
-	return 0;
-}
-
-jboolean JNICALL
-Java_jdk_jfr_internal_JVM_emitEvent(JNIEnv *env, jclass clazz, jlong eventTypeId, jlong timestamp, jlong periodicType)
+Java_jdk_jfr_internal_JVM_endRecording(JNIEnv *env, jobject obj)
 {
 	// TODO: implementation
-	return JNI_FALSE;
 }
 
 jobject JNICALL
-Java_jdk_jfr_internal_JVM_getAllEventClasses(JNIEnv *env, jclass clazz)
+Java_jdk_jfr_internal_JVM_getAllEventClasses(JNIEnv *env, jobject obj)
 {
 	// TODO: implementation
 	return NULL;
 }
 
 jlong JNICALL
-Java_jdk_jfr_internal_JVM_getUnloadedEventClassCount(JNIEnv *env, jclass clazz)
+Java_jdk_jfr_internal_JVM_getUnloadedEventClassCount(JNIEnv *env, jobject obj)
 {
 	// TODO: implementation
 	return 0;
 }
 
 jlong JNICALL
-Java_jdk_jfr_internal_JVM_getClassId(JNIEnv *env, jclass clz, jobject clazz)
+Java_jdk_jfr_internal_JVM_getClassId(JNIEnv *env, jclass cls, jclass clazz)
```

Comment:
I think `clazz` should be used consistently for the context class; the second `jclass` parameter should get a new name, like "subjectClass" or "targetClass" that clarifies its purpose.